src/java.base/share/classes/jdk/internal/reflect/MethodHandleAccessorFactory.java

```diff
@@ -101,23 +106,63 @@ static ConstructorAccessorImpl newConstructorAccessor(Constructor<?> ctor) {
         // Ensure class initialized outside the invocation of method handle
         // so that EIIE is propagated (not wrapped with ITE)
         ensureClassInitialized(ctor.getDeclaringClass());
+        try {
+            MethodHandle target = makeConstructorHandle(JLIA.unreflectConstructor(ctor));
+            return DirectConstructorHandleAccessor.constructorAccessor(ctor, target);
+        } catch (IllegalAccessException e) {
+            throw new InternalError(e);
+        }
+    }
+
+    /**
+     * Creates a ConstructorAccessor that is capable of creating instances
+     * of the given class and instantiated by the given constructor.
+     *
+     * @param decl the class to instantiate
+     * @param ctor the constructor to call
+     * @return an accessible constructor
+     */
+    static ConstructorAccessorImpl newSerializableConstructorAccessor(Class<?> decl, Constructor<?> ctor) {
+        if (!constructorInSuperclass(decl, ctor)) {
+            throw new UnsupportedOperationException(ctor + " not a superclass of " + decl.getName());
```

Comment:
Notice in JShell 20, you could do this:
```
jshell> import sun.reflect.ReflectionFactory;
jshell> var rf = ReflectionFactory.getReflectionFactory()
jshell> var c = rf.newConstructorForSerialization(String.class, Boolean.class.getConstructor(boolean.class))
jshell> String d = (String) c.newInstance(false)
jshell> d.length()
```
which ends up in an NPE for `d.value` is null. This UOE is technically a new behavior that should be mentioned in the CSR.

Reply:
good catch.  will update the CSR.